make/clang-tidy

```diff
@@ -0,0 +1,16 @@
+
+tidy_base_checks=*perf*,*modernize*,*misc*,*google*,*readability*,*optin*
+modern_excludes=,-modernize-replace-random-shuffle,-modernize-use-override,-modernize-concat-nested-namespaces,-modernize-avoid-c-arrays
+readability_excludes=,-readability-magic-numbers,-readability-convert-member-functions-to-static,-readability-inconsistent-declaration-parameter-name,-readability-function-size,-readability-named-parameter,-readability-container-size-empty,-readability-implicit-bool-conversion,-readability-else-after-return,-readability-container-size-empty
+misc_excludes=,-misc-non-private-member-variables-in-classes,-misc-unused-parameters,-misc-unused-using-decls,-misc-definitions-in-headers,-misc-unconventional-assign-operator
+google_excludes=,-google-runtime-references,-google-global-names-in-headers,-google-build-using-namespace,-google-runtime-int
+things_we_should_look_at=,-google-readability-casting,-google-explicit-constructor,-modernize-use-using,-modernize-pass-by-value,-readability-implicit-bool-conversion,-readability-const-return-type,-readability-avoid-const-params-in-decls,-modernize-return-braced-init-list,-modernize-use-using
+tidy_checks?= $(tidy_base_checks) $(modern_excludes) $(readability_excludes) $(misc_excludes) $(google_excludes) $(things_we_should_look_at)
+
+.PHONY: clang-tidy
+clang-tidy:
```

Comment:
So... we might be able to override a variable here just for this target. Before these lines, try adding:

```
clang-tidy: CXX=clang++
```

If this works, great! If it doesn't, it's probably some interaction with phony targets and rules for them.

Reply:
Turns out I'm dumb and g++ works just fine haha